chemprop/utils/mixins.py

```diff
@@ -4,17 +4,47 @@
 
 class ReprMixin:
     def __repr__(self) -> str:
-        items = self.get_params()
+        default_params = self._get_default_params()
+        param_repr_list = [
+            f"{name}={repr(value)}"
+            for name, value in self.get_params(deep=False).items()
+            if value != default_params[name]
+        ]
 
-        if len(items) > 0:
-            keys, values = zip(*items)
-            sig = inspect.signature(self.__class__)
-            defaults = [sig.parameters[k].default for k in keys]
-            items = [(k, v) for k, v, d in zip(keys, values, defaults) if v != d]
+        return f"{self.__class__.__name__}({','.join(param_repr_list)})"
 
-        argspec = ", ".join(f"{k}={repr(v)}" for k, v in items)
+    @classmethod
+    def _get_default_params(cls) -> dict[str, Any]:
+        """Get the default parameters for initializing the class."""
+        sig = inspect.signature(cls)
+        return {k: v.default for k, v in sig.parameters.items()}
 
-        return f"{self.__class__.__name__}({argspec})"
+    def _get_param_names(self) -> list[str]:
+        """Get the parameter names for initializing the class."""
+        return sorted(self._get_default_params().keys())
 
-    def get_params(self) -> Collection[tuple[str, Any]]:
-        return self.__dict__.items()
+    def get_params(self, deep: bool = True) -> dict[str, Any]:
+        """Get the current parameters of the class, required to initialize the class.
+
+        This method mimics the behavior of :method:`sklearn.base.BaseEstimator.get_params`.
+
+        Parameters
+        ----------
+        deep : bool, default=True
+            If True, will return the parameters for this estimator and
+            contained subobjects that are estimators.
```

Comment:
Is this correct?

Reply:
This is copied from [scikit-learn](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/base.py#L234). I see the function checks if the contained thing `hasattr(value, "get_params") and not isinstance(value, type)`. Perhaps those two conditions mean the contained thing is an estimator in scikit-learn. 

But in our case, `get_params()` is only implemented for the loss functions, which aren't estimators (based on my understanding of the scikit-learn [docs](https://scikit-learn.org/stable/developers/develop.html)).

Reply:
It's clear this is copied from sklearn because we have no concept of an `Estimator` in our code

Reply:
Then perhaps a more correct description would be "If True, will return the parameters for this object and all contained subobjects."

Reply:
updated the docstring